app/services/concerns/suspend_silence_post_action.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module SuspendSilencePostAction
```

Comment:
I think a better fit would be to make this an "action" class rather than a concern, see https://github.com/discourse/discourse/blob/11369018b6841b7afa1a564858cbefb710b05be9/plugins/chat/app/services/chat/action/publish_auto_removed_user.rb . It's very similar but it's more obvious in the service when you are calling `SuspendSilencePostAction.call(guardian: guardian)` rather than calling `perform_post_action` which doesn't appear in the class